src/twisted/internet/_sslverify.py

```diff
@@ -1747,32 +1747,23 @@ def _verifyCallback(conn, cert, errno, depth, preverify_ok):
 
 
 @implementer(ICipher)
-class OpenSSLCipher(FancyEqMixin, object):
+@attr.s(frozen=True)
+class OpenSSLCipher(object):
     """
     A representation of an OpenSSL cipher.
-    """
-    compareAttributes = ('fullName',)
-
-    def __init__(self, fullName):
-        """
-        @param fullName: The full name of the cipher. For example
-            C{u"ECDHE-RSA-AES256-GCM-SHA384"}.
-        @type fullName: L{unicode}
-        """
-        self.fullName = fullName
-
 
-    def __repr__(self):
-        """
-        A runnable representation of the cipher.
-        """
-        return 'OpenSSLCipher({0!r})'.format(self.fullName)
+    @ivar fullName: The full name of the cipher. For example
+        C{u"ECDHE-RSA-AES256-GCM-SHA384"}.
+    @type fullName: L{unicode}
+    """
+    fullName = attr.ib()
 
 
 
+@lru_cache(maxsize=32)
 def _expandCipherString(cipherString, method, options):
     """
-    Expand C{cipherString} according to C{method} and C{options} to a list
+    Expand C{cipherString} according to C{method} and C{options} to a tuple
```

Comment:
As we annotate with types, this should (at some future point) probably be a `Sequence[ICipher]`, not a `Tuple`.

Reply:
`Sequence` doesn't imply `Hashable`, though. That means... mixing the two? Or a custom Protocol? I am not up to speed on this MyPy stuff.